src/examples/environment_renderer.cpp

```diff
@@ -0,0 +1,657 @@
+      , vertexShader_(0)
+      , fragmentShader_(0)
+      , vao_(0)
+      , vbo_(0)
+      , cubeMapTexture_(0)
+      , viewMatrixLocation_(-1)
+      , projectionMatrixLocation_(-1)
+      , cubeMapLocation_(-1)
+  {
+  }
+
+  EnvironmentRenderer::~EnvironmentRenderer()
+  {
+    shutdown();
+  }
+
+  bool EnvironmentRenderer::initialize(const string &cubeMapPath)
+  {
+    if (initialized_)
+    {
+      return true;
+    }
+
+    cout << "Initializing Environment Renderer..." << endl;
+
+    // Create shaders
+    createShaders();
+    if (shaderProgram_ == 0)
+    {
+      cerr << "Failed to create environment shaders" << endl;
+      return false;
+    }
+
+    // Create cube geometry
+    createCubeGeometry();
+    if (vao_ == 0)
+    {
+      cerr << "Failed to create cube geometry" << endl;
+      return false;
+    }
+
+    // Load cube map texture
+    if (!cubeMapPath.empty())
+    {
+      if (!loadCubeMap(cubeMapPath))
+      {
+        cout << "Failed to load cube map from " << cubeMapPath << ", using procedural environment" << endl;
+        createProceduralCubeMap();
+      }
+    }
+    else
+    {
+      createProceduralCubeMap();
+    }
+
+    // Get uniform locations
+    viewMatrixLocation_ = glGetUniformLocation(shaderProgram_, "view");
+    projectionMatrixLocation_ = glGetUniformLocation(shaderProgram_, "projection");
+    cubeMapLocation_ = glGetUniformLocation(shaderProgram_, "skybox");
+
+    initialized_ = true;
+    cout << "Environment Renderer initialized successfully" << endl;
+    return true;
+  }
+
+  void EnvironmentRenderer::render(const glm::mat4 &viewMatrix, const glm::mat4 &projectionMatrix)
+  {
+    if (!initialized_ || !enabled_ || !hasCubeMapTexture_)
+    {
+      return;
+    }
+
+    // Disable depth writing (but keep depth testing to ensure skybox is behind everything)
+    glDepthMask(GL_FALSE);
+
+    glUseProgram(shaderProgram_);
+
+    // Set uniforms
+    glUniformMatrix4fv(viewMatrixLocation_, 1, GL_FALSE, glm::value_ptr(viewMatrix));
+    glUniformMatrix4fv(projectionMatrixLocation_, 1, GL_FALSE, glm::value_ptr(projectionMatrix));
+
+    // Bind cube map texture
+    glActiveTexture(GL_TEXTURE0);
+    glBindTexture(GL_TEXTURE_CUBE_MAP, cubeMapTexture_);
+    glUniform1i(cubeMapLocation_, 0);
+
+    // Render cube
+    glBindVertexArray(vao_);
+    glDrawArrays(GL_TRIANGLES, 0, 36);
+    glBindVertexArray(0);
+
+    // Re-enable depth writing
+    glDepthMask(GL_TRUE);
+
+    glUseProgram(0);
+  }
+
+  void EnvironmentRenderer::shutdown()
+  {
+    destroyResources();
+    initialized_ = false;
+  }
+
+  bool EnvironmentRenderer::loadCubeMap(const string &filePath)
+  {
+    // Check if path is a directory
+    struct stat pathStat;
+    if (stat(filePath.c_str(), &pathStat) == 0)
+    {
+      if (S_ISDIR(pathStat.st_mode))
+      {
+        // Path is a directory
+        return loadDirectoryCubeMap(filePath);
+      }
+      else if (S_ISREG(pathStat.st_mode))
```

Comment:
Now it should only support the directory to load environment's cube map, please remove the case.